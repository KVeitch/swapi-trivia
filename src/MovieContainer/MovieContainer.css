.movie__container { 
    background-color: rgba(67, 52, 45, 0.7);
    color: white;
    display: flex;
    height: 53vh;
    margin-left: 7%;
    width: 85%;
    overflow:scroll;
}