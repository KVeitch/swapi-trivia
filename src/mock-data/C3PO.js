const insults = [
    {quote: 'Don’t call me a mindless philosopher, you overweight glob of grease!'},
    {quote: 'After all, he’s only a Wookiee.'},
    {quote: 'I can’t abide these Jawas. Disgusting creatures.'},
    {quote: 'You’ll be malfunctioning within a day, you nearsighted scrap pile.'},
    {quote: 'Don’t worry about Master Luke. I’m sure he’ll be all right. He’s quite clever, you know… for a human being.'}
]

export default insults;