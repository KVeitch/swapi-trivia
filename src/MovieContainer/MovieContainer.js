import React from 'react';
import MovieCard from '../MovieCard/MovieCard';
import PropTypes from 'prop-types';
import './MovieContainer.css';
import loading from '../images/loading.gif';

const MovieContainer = props => {
  if(props.movies.length === 0){
    return(
      <>
      <h1>Loading..</h1>
      <div className='div__image'>
        <p className='loading__text'>Please be patient,</p>
        <p className='loading__text'>the service droids are doing their best.</p>
        <img className='loading__image' src={loading} alt="BB-8 droid rolling back and forth" />
      </div>
      </>
    )
  }

  const movieMap = [4,5,6,1,2,3,7]
  const movieList = props.movies.map((movie, i) => 
      <MovieCard 
        title={movie.title}
        episode={movie.episode_id}
        movieImage={movie.movieImage}
        key={movie.episode_id}
        releaseYear={movie.release_date}
        characters={movie.characters}
        changeSelectedMovie={props.changeSelectedMovie}
        setCurrentCharacters={props.setCurrentCharacters}
        id={movieMap[i]}
        />)

  return (
    <section className='movie__container'>
      {movieList}
    </section>
  )
}

export default MovieContainer;

MovieContainer.propTypes ={
  movies:PropTypes.array,
  setCurrentCharacters:PropTypes.func,
  changeSelectedMovie:PropTypes.func, 
  characters:PropTypes.array
}