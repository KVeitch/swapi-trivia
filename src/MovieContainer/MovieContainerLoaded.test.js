import React from 'react';
import { shallow } from 'enzyme';
import MovieContainer from '../MovieContainer/MovieContainer';
import { sortedMovies } from '../mock-data/movie-data'

describe('Movie Container Loaded', () => {

  it('should match the snapshot with all the data passed in correctly', () => {
      const wrapper =
          shallow(<MovieContainer
                  movies={sortedMovies}
                  changeSelectedMovie={jest.fn()}/>)
      expect(wrapper).toMatchSnapshot();
  });
});