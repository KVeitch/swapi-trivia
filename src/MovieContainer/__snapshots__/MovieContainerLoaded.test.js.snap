// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Movie Container Loaded should match the snapshot with all the data passed in correctly 1`] = `
<section
  className="movie__container"
>
  <MovieCard
    changeSelectedMovie={[MockFunction]}
    characters={
      Array [
        "https://swapi.co/api/people/2/",
        "https://swapi.co/api/people/3/",
        "https://swapi.co/api/people/10/",
        "https://swapi.co/api/people/11/",
        "https://swapi.co/api/people/16/",
        "https://swapi.co/api/people/20/",
        "https://swapi.co/api/people/21/",
        "https://swapi.co/api/people/32/",
        "https://swapi.co/api/people/33/",
        "https://swapi.co/api/people/34/",
        "https://swapi.co/api/people/36/",
        "https://swapi.co/api/people/35/",
      ]
    }
    episode={1}
    id={4}
    key="1"
    releaseYear="1999-05-19"
    title="The Phantom Menace"
  />
  <MovieCard
    changeSelectedMovie={[MockFunction]}
    characters={
      Array [
        "https://swapi.co/api/people/2/",
        "https://swapi.co/api/people/3/",
        "https://swapi.co/api/people/6/",
        "https://swapi.co/api/people/7/",
        "https://swapi.co/api/people/10/",
        "https://swapi.co/api/people/11/",
        "https://swapi.co/api/people/20/",
        "https://swapi.co/api/people/21/",
        "https://swapi.co/api/people/22/",
        "https://swapi.co/api/people/33/",
        "https://swapi.co/api/people/36/",
        "https://swapi.co/api/people/35/",
      ]
    }
    episode={2}
    id={5}
    key="2"
    releaseYear="2002-05-16"
    title="Attack of the Clones"
  />
  <MovieCard
    changeSelectedMovie={[MockFunction]}
    characters={
      Array [
        "https://swapi.co/api/people/1/",
        "https://swapi.co/api/people/2/",
        "https://swapi.co/api/people/3/",
        "https://swapi.co/api/people/4/",
        "https://swapi.co/api/people/5/",
        "https://swapi.co/api/people/6/",
        "https://swapi.co/api/people/7/",
        "https://swapi.co/api/people/10/",
        "https://swapi.co/api/people/11/",
        "https://swapi.co/api/people/12/",
        "https://swapi.co/api/people/13/",
        "https://swapi.co/api/people/35/",
      ]
    }
    episode={3}
    id={6}
    key="3"
    releaseYear="2005-05-19"
    title="Revenge of the Sith"
  />
  <MovieCard
    changeSelectedMovie={[MockFunction]}
    characters={
      Array [
        "https://swapi.co/api/people/1/",
        "https://swapi.co/api/people/2/",
        "https://swapi.co/api/people/3/",
        "https://swapi.co/api/people/4/",
        "https://swapi.co/api/people/5/",
        "https://swapi.co/api/people/6/",
        "https://swapi.co/api/people/7/",
        "https://swapi.co/api/people/8/",
        "https://swapi.co/api/people/9/",
        "https://swapi.co/api/people/10/",
        "https://swapi.co/api/people/12/",
      ]
    }
    episode={4}
    id={1}
    key="4"
    releaseYear="1977-05-25"
    title="A New Hope"
  />
  <MovieCard
    changeSelectedMovie={[MockFunction]}
    characters={
      Array [
        "https://swapi.co/api/people/1/",
        "https://swapi.co/api/people/2/",
        "https://swapi.co/api/people/3/",
        "https://swapi.co/api/people/4/",
        "https://swapi.co/api/people/5/",
        "https://swapi.co/api/people/10/",
        "https://swapi.co/api/people/13/",
        "https://swapi.co/api/people/14/",
        "https://swapi.co/api/people/18/",
        "https://swapi.co/api/people/20/",
        "https://swapi.co/api/people/21/",
      ]
    }
    episode={5}
    id={2}
    key="5"
    releaseYear="1980-05-17"
    title="The Empire Strikes Back"
  />
  <MovieCard
    changeSelectedMovie={[MockFunction]}
    characters={
      Array [
        "https://swapi.co/api/people/1/",
        "https://swapi.co/api/people/2/",
        "https://swapi.co/api/people/3/",
        "https://swapi.co/api/people/4/",
        "https://swapi.co/api/people/5/",
        "https://swapi.co/api/people/10/",
        "https://swapi.co/api/people/13/",
        "https://swapi.co/api/people/14/",
        "https://swapi.co/api/people/16/",
        "https://swapi.co/api/people/18/",
        "https://swapi.co/api/people/20/",
        "https://swapi.co/api/people/21/",
      ]
    }
    episode={6}
    id={3}
    key="6"
    releaseYear="1983-05-25"
    title="Return of the Jedi"
  />
  <MovieCard
    changeSelectedMovie={[MockFunction]}
    characters={
      Array [
        "https://swapi.co/api/people/1/",
        "https://swapi.co/api/people/3/",
        "https://swapi.co/api/people/5/",
        "https://swapi.co/api/people/13/",
        "https://swapi.co/api/people/14/",
        "https://swapi.co/api/people/27/",
        "https://swapi.co/api/people/84/",
        "https://swapi.co/api/people/85/",
        "https://swapi.co/api/people/86/",
        "https://swapi.co/api/people/87/",
        "https://swapi.co/api/people/88/",
      ]
    }
    episode={7}
    id={7}
    key="7"
    releaseYear="2015-12-11"
    title="The Force Awakens"
  />
</section>
`;
